lib/manageiq/providers/workflows/runner.rb

```diff
@@ -0,0 +1,86 @@
+module ManageIQ
+  module Providers
+    module Workflows
+      class Runner
+        include Vmdb::Logging
+
+        class << self
+          def runner
+            @runner ||= new.tap(&:start)
+          end
+        end
+
+        attr_reader :workflows
+
+        def initialize
+          require "floe"
+          require "concurrent/hash"
+
+          @workflows          = Concurrent::Hash.new
+          @docker_wait_thread = nil
+        end
+
+        def start
+          $workflows_log.debug("Runner: Starting workflows runner...")
+          self.docker_wait_thread = Thread.new { docker_wait }
+          $workflows_log.debug("Runner: Starting workflows runner...Complete")
+        end
+
+        def stop
+          $workflows_log.debug("Runner: Stopping workflows runner...")
+          stop_thread(docker_wait_thread)
+
+          self.docker_wait_thread = nil
+          $workflows_log.debug("Runner: Stopping workflows runner...Complete")
+        end
+
+        def add_workflow(workflow, queue_args)
+          workflows[workflow.id] ||= [workflow, queue_args]
+        end
+
+        def delete_workflow(workflow)
+          workflows.delete(workflow.id)
+        end
+
+        private
+
+        attr_accessor :docker_wait_thread
+
+        def docker_wait
+          loop do
+            docker_runner = Floe::Runner.for_resource("docker")
+            docker_runner.wait do |event, runner_context|
```

Comment:
feeling like we only care about `:delete`

Reply:
Yes I agree this was "an abundance of caution" where we poke the queue more often that we have to, but without accidentally leaving a workflow "hung"

Reply:
yea. I worry about not coding the exact list of "done" and therefore not reacting because someone said "fire" in the wrong language

Reply:
If we have 10 parallel tasks, would be nice to know which Task to check.

Reply:
I played with checking if it was `ready?` before putting the task on miq_queue which is essentially what step_nonblock_ready? will check anyway when we go to step maybe we go that direction (follow-up? this will work but less efficient)